src/impl/tpls/KokkosSparse_spmv_tpl_spec_decl.hpp

```diff
@@ -44,9 +44,214 @@
 #ifndef KOKKOSPARSE_SPMV_TPL_SPEC_DECL_HPP_
 #define KOKKOSPARSE_SPMV_TPL_SPEC_DECL_HPP_
 
+// cuSPARSE
+#ifdef KOKKOSKERNELS_ENABLE_TPL_CUSPARSE
+#include "cusparse.h"
+
 namespace KokkosSparse {
 namespace Impl {
-}
-}
+
+  template <class AMatrix, class XVector, class YVector>
+  void spmv_cusparse(const char mode[],
+		     typename YVector::non_const_value_type const & alpha,
+		     const AMatrix& A,
+		     const XVector& x,
+		     typename YVector::non_const_value_type const & beta,
+		     const YVector& y) {
+    using offset_type  = typename AMatrix::non_const_size_type;
+    // using ordinal_type = typename AMatrix::non_const_ordinal_type;
+    using value_type   = typename AMatrix::non_const_value_type;
+
+#if defined(CUSPARSE_VERSION) && (10300 <= CUSPARSE_VERSION)
```

Comment:
I thought it was CUDA 10.1 that added the new interface.

Reply:
I think it's a newer "patch" update to 10.1 that added the new interface, 10.1.patch, but is not included in all versions of 10.1 (in particular not the version we have a sems module for nightly testing)

Reply:
It's a bit complicated, but the bottom line is that cuda and cusparse do not necessarily have the same version numbering.

| Release number | CUDA_VERSION | CUSPARSE_VERSION
|---|---|---|
| 10.1.168 | 10100 | undef |
| 10.1.243 | 10100 | 10300 |

Note that in release 10.1.168, cusparse does define the following macros:
```
#define CUSPARSE_VER_MAJOR 10
#define CUSPARSE_VER_MINOR 2
#define CUSPARSE_VER_PATCH 0
#define CUSPARSE_VER_BUILD 0
```
where as in 10.1.243 it defines
```
#define CUSPARSE_VER_MAJOR 10
#define CUSPARSE_VER_MINOR 3
#define CUSPARSE_VER_PATCH 0
#define CUSPARSE_VER_BUILD 243
#define CUSPARSE_VERSION (CUSPARSE_VER_MAJOR * 1000 + \
                          CUSPARSE_VER_MINOR *  100 + \
                          CUSPARSE_VER_PATCH)
```